### ISSUE TITLE
SHA-256 digest error for jar file

### ISSUE THREAD

**Issue by b4git** [Col-E/Recaf#378]
A JAR file was decompiled, modified and exported using recaf, however after running the JAVA program with the modified JAR it threw the SHA-256 digest error for jar file error.
What is the proper way to handle this?

**Comment by Col-E** [comment]
You can't make an edit to a class and expect the class's bytecode hash to stay the same. So your only option is to remove the hash validation. Doing so in a jar file is quite easy:

1. Delete the SF/RSA files in `META-INF`
2. Modify the `MANIFEST.MF` file in `META-INF` to remove the `Name` and `SHA-256-Digest` entry of the files you modify
    - Its easiest to just delete all the entries, there is no real repercussions for doing this

**Comment by b4git** [comment]
@Col-E  Sure that  was the way I was following to solve the problem temporarily, but I  was unsure whether it was possible to specify to recaf to unsign/sign/resign/regenerate/replace the JAR files, and it seems from your response we have to do it manually. Thanks for verifying.

**Comment by Col-E** [comment]
Yeah, Recaf focuses on the bytecode and doesn't update things like `MANIFEST.MF` entries atm. Although #381 does add manifest updating for remapped main classes. Its not out of scope per say, just not something I've personally invested time into.

**Comment by Eboubaker** [comment]
Thank you @Col-E I was about to lose hope.

**Comment by Eboubaker** [comment]
@Col-E 
Also an improvement suggestion for the program before exporting to check if any RSA signature or similar thing exist and warn you about this problem(maybe even remove the entries in META-INF automatically with a single click of a button)
